### ISSUE TITLE
Icons for Appendix B could be visually improved

### ISSUE THREAD

**Issue by azaroth42** [IIIF/api#745]
From #739, the requirement icons could be improved. In particular:
- The red of required looks like a No (and hence N/A)
- The grey optional and the grey not allowed are too similar

Proposals welcome for a new set of icons.

**Comment by jpstroop** [comment]
Tables / icons in question: http://iiif.io/api/presentation/2.1/#b-summary-of-metadata-requirements

**Comment by zimeon** [comment]
Additional requirement:

  * Shapes/icons must work even without the colors

Hard to think what colors evoke required vs optional...

**Comment by axamei** [comment]
How about the following shapes/icons... 

Required: Lock
Recommended: Thumbs up
Optional: O
Not Allowed: X

**Comment by mikeapp** [comment]
Agree with not requiring color to differentiate/be understood.

**Comment by azaroth42** [comment]
:+1: to @axamei's suggestions :)

**Comment by jpstroop** [comment]
So, when I fill out a form w/ required fields, they usually have some kind of "*" in them. 

:eight_spoked_asterisk:

**Comment by jpstroop** [comment]
(That's the only one I really wonder about)

**Comment by azaroth42** [comment]
Alternative from @zimeon: ! + ? X

**Comment by jpstroop** [comment]
I think "!" is confusing. To me that says no, and if anything, "X" means "check, yes, you must fill this box"

**Comment by zimeon** [comment]
Or could use * in place of ! for required, following @jpstroop's comment

**Comment by jpstroop** [comment]
As Mike suggested offilne: http://ux.stackexchange.com/a/41548 ?? Pretty boring.

**Comment by jpstroop** [comment]
(and by boring, I mean, they sorta make sense)

**Comment by zimeon** [comment]
I kinda feel the disc/half/circle has been used in something I've been involved with before but I can't find it . I note that for [ResourceSync](http://www.openarchives.org/rs/1.0/resourcesync#tab_4) we ended up with words and 'X' for not allowed.

**Comment by tomcrane** [comment]
I like the boring ones in http://ux.stackexchange.com/a/41548, plus X for not allowed.

...and they are in Unicode!

● ◓ ○ X

... although unfortunately not the same size. That's annoying.

https://en.wikipedia.org/wiki/Geometric_Shapes

**Comment by jpstroop** [comment]
Looking through the Unicode DB...

For "x" there's also:
 * 𝖷 (U+1D5B7)
 * Ⓧ (U+24CD) 
 * ✕ (U+2715) 
 * ✖ (U+2716)
 * 𝗫 (U+1D5EB)
 * ☠ (U+2620) :wink: 

For "o" there's:
 * 🞅 (U+1F785)
 * 🞆 (U+1F786)

For a bigger solid "o" theres:
 * ⏺ (U+23FA )

That said, I don't know what the usability implications are for using arbitrary Unicode characters vs. images that can have `alt` labels, but I'm guessing that the latter is better.

**Comment by azaroth42** [comment]
Images with alts would be a more consistent experience.  I see [] for the two "o"s above, for example.

**Comment by jpstroop** [comment]
:+1:

**Comment by zimeon** [comment]
:1: on image + alts, how do we move forward on the overall question of which icons??

**Comment by jpstroop** [comment]
I think the reason for our differences has to do with perspective. If you look at the tables as a set of requirements, you want to see a set of symbols that reflect what is allowed--in this scenario, e.g. '!" means "not allowed" to me and "*" looks like something that's required.

However, if you look to the tables for validation, I think you have different expectations, and e.g. "!" looks like a something that's required and you have it wrong if it's not there, and "*" looks like something optional.

Does this make sense to anyone else? If so, maybe we first need to decide on a perspective (the first para above is definitely mine) and go from there?

**Comment by mikeapp** [comment]
:+1: on images with alt tags.  I like the circles because the amount of fill suggests relative levels of importance, rather than the `!` which, as Jon notes, can be read as 'required!' or 'problem!' depending on your perspective.

**Comment by jpstroop** [comment]
> `!` which, as Jon notes, can be read as 'required!' or 'problem!' depending on your perspective.

Much more eloquent and concise. Yes, that. :wink:

**Comment by zimeon** [comment]
I would also be fine with the circles, perhaps they span both of the perspectives @jpstroop describes? FWIW, I look at the table with the requirements perspective first

**Comment by tomcrane** [comment]
I like the circles plus X, and the X to me means "you can't do this here".

Can we use iiif-discuss for "hallway usability testing"? There are lots of new implementers on there right now.

**Comment by jpstroop** [comment]
☐ ☑ ☒

?

**Comment by azaroth42** [comment]
To confirm, we believe that this issue can be addressed at any time and shouldn't be considered a blocker for 2.1, right?

**Comment by azaroth42** [comment]
Can we suggest someone that would be motivated to come up with a range of options for discussion?

**Comment by tomcrane** [comment]
IIIF AV Proposal is https://github.com/IIIF/api/issues/745#issuecomment-192009547, with green for the circles and red for the X.

Just change the linked images.

**Comment by zimeon** [comment]
It seems that we will, by default, change the images in all existing specs because they all use the same links with `/img/metadata-api/...` to:

```
source/img/metadata-api/not_allowed.png
source/img/metadata-api/optional.png
source/img/metadata-api/recommended.png
source/img/metadata-api/required.png
```

I think this is fine (though I don't much like the `metadata-api` in the path).

**Comment by azaroth42** [comment]
Closed by #1475
